Source/Core/Core/IOS/USB/LibusbDevice.cpp

```diff
@@ -124,28 +124,40 @@ std::string LibusbDevice::GetErrorName(const int error_code) const
   return libusb_error_name(error_code);
 }
 
-bool LibusbDevice::Attach(const u8 interface)
+bool LibusbDevice::Attach()
 {
-  if (m_device_attached && interface != m_active_interface)
-    return ChangeInterface(interface) == 0;
-
+  // Already opened and an interface is attached -- nothing to do.
   if (m_device_attached)
     return true;
 
-  m_device_attached = false;
-  NOTICE_LOG(IOS_USB, "[%04x:%04x] Opening device", m_vid, m_pid);
-  const int ret = libusb_open(m_device, &m_handle);
-  if (ret != 0)
+  if (!m_handle)
   {
-    ERROR_LOG(IOS_USB, "[%04x:%04x] Failed to open: %s", m_vid, m_pid, libusb_error_name(ret));
-    return false;
+    NOTICE_LOG(IOS_USB, "[%04x:%04x] Opening device", m_vid, m_pid);
+    const int ret = libusb_open(m_device, &m_handle);
+    if (ret != 0)
+    {
+      ERROR_LOG(IOS_USB, "[%04x:%04x] Failed to open: %s", m_vid, m_pid, libusb_error_name(ret));
+      m_handle = nullptr;
+      return false;
+    }
   }
-  if (AttachInterface(interface) != 0)
+  if (AttachInterface(0) != 0)
```

Comment:
Had to look twice to notice that `0` (or any other value) is necessary here now since the `interface` parameter is gone. But is zero really a good value for this? Feels like it should either just fail, or pick some sort of default that might be indistinguishable from magic. Can't tell which one it is, or whether it's a third option.

Reply:
Hmmm yeah, it might be better to just not claim any interface by default since the function is supposed to just attach the device... I'll change this tomorrow

Reply:
I've changed the way interfaces are claimed completely (as part of some bug fixes). Now all interfaces are claimed ahead-of-time as it's difficult to predict which interfaces are going to be used and errorprone to change them when needed. Claiming them ahead-of-time ensures the emulated software will always be able to use the interface it wants, just like on an actual Wii.